### ISSUE TITLE
Training a 70B Reward Model on a single node with 8 H100 96GB

### ISSUE THREAD

**Issue by alexnikulkov** [huggingface/trl#3218]
I'm getting an OOM even with the most memory-efficient settings I could come up with.
To save memory I'm using SGD optimizer, which doesn't have any state.
With bf16 dtype, I'd expect to need (2+2)*70/8=35 GB of memory per device (according to [source](https://huggingface.co/spaces/nanotron/ultrascale-playbook?section=zero-3:_adding_parameter_partitioning)), yet it OOMs on my 96GB GPUs.
I'm running the [reward model training example from trl](https://github.com/huggingface/trl/blob/main/examples/scripts/reward_modeling.py) with command:
`accelerate launch reward_modeling.py  --model_name_or_path meta-llama/Meta-Llama-3-70B-Instruct   --dataset_name trl-lib/ultrafeedback_binarized     --output_dir fsdp_test     --per_device_train_batch_size 1     --num_train_epochs 1     --gradient_checkpointing False     --learning_rate 1.0e-5     --logging_steps 25     --eval_strategy steps     --eval_steps 50     --max_length 1024 --torch_dtype bfloat16 --optim sgd`

My FSDP config is:
```
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: FSDP
downcast_bf16: yes
enable_cpu_affinity: false
fsdp_config:
  fsdp_activation_checkpointing: false # broken
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
  fsdp_backward_prefetch: NO_PREFETCH
  fsdp_cpu_ram_efficient_loading: true
  fsdp_forward_prefetch: false
  fsdp_offload_params: true
  fsdp_sharding_strategy: FULL_SHARD
  fsdp_state_dict_type: SHARDED_STATE_DICT
  fsdp_sync_module_states: true
  fsdp_use_orig_params: true
machine_rank: 0
main_training_function: main
mixed_precision: bf16
num_machines: 1
num_processes: 8
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

**Comment by qgallouedec** [comment]
You should try with gradient_checkpointing
